lib/progressbar.c

```diff
@@ -54,7 +55,12 @@ progressbar *progressbar_new(char *label, unsigned long max)
 
 void progressbar_update_label(progressbar *bar, char *label)
 {
-  bar->label = label;
+  if (bar->label) {
+    free(bar->label);
+  }
+  bar->label = calloc(strlen(label)+1, sizeof(char));
+  label = strncpy(bar->label, label, strlen(label));
```

Comment:
@marcomorain thanks for the suggestion, I will make the changes.

@doches don't merge yet.